src/librustc/ty/maps.rs

```diff
@@ -512,6 +513,15 @@ impl<'tcx> QueryDescription for queries::extern_crate<'tcx> {
     }
 }
 
+// If enabled, send a message to the profile-queries thread
+macro_rules! profq_msg {
+    ($tcx:expr, $msg:expr) => {
+        if $tcx.sess.opts.debugging_opts.profile_queries {
```

Comment:
Could you wrap this `if` in `if cfg!(debug_assertions) { ... }`? Then it will be filtered out at compile time if debug assertions are disabled.